easyrest/models/validator.py

```diff
@@ -29,3 +31,98 @@ def validation(schema, data):
             log.error('%s', error.message)
         raise ValidationError(data)
     return True
+
+
+def check_action_access(user_role, foreign_role, action):
+    """This function checks the possibility of the specified action by the user.
+
+       The function checks the rights of the current user's role for other roles.
+       access_list = {
+        'current_user_role': {
+            'other_user_role1': ['action1', 'action2'],
+            'other_user_role2': [],
+            'other_user_role3': [],
+            ...
+
+    :param (str) user_role: User role that requests action
+    :param (str) foreign_role: Requested role
+    :param (str) action: Action name
+    :return: True if action is allowed
+    :raise HTTPForbidden: If the action is prohibited
+    :raise HTTPNotFound: If the action is not found
+    """
+    log = logging.getLogger(__name__)
+    user_access_list = {
+        'Client': {
+            'Client': ['create'],
+            'Waiter': [],
+            'Administrator': [],
+            'Owner': [],
+            'Moderator': [],
+            'Admin': []
+        },
+        'Waiter': {
+            'Client': ['create'],
```

Comment:
Is this normal, when Waiter can create client? Or by "create" you mean other action (besides client creation/modification)?

Reply:
> Is this normal, when Waiter can create client?

I think yes. Create client == Sign Up. Any role can create a user with the role "Client".
I tried to make this access list flexible, so I did not focus on such things. I expected that I would customize actions as further developed. So this is not the final version.